hpy/devel/src/runtime/argparse.c

```diff
@@ -94,75 +214,95 @@ HPyAPI_RUNTIME_FUNC(int)
 HPyArg_ParseKeywords(HPyContext ctx, HPy *args, HPy_ssize_t nargs, HPy kw,
                      const char *fmt, const char *keywords[], ...)
 {
-  va_list vl;
-  va_start(vl, keywords);
-  const char *fmt1 = fmt;
-  int optional = 0;
-  int keyword_only = 0;
-  HPy_ssize_t i = 0;
-  HPy_ssize_t nkw = 0;
-  HPy current_arg;
-
-  // first count positional only arguments
-  while (keywords[nkw] != NULL && !*keywords[nkw]) nkw++;
-  // then check and count the rest
-  while (keywords[nkw] != NULL) {
-    if (!*keywords[nkw]) {
-      HPyErr_SetString(ctx, ctx->h_TypeError, "XXX: Empty keyword parameter name");
-      va_end(vl);
-      return 0;
+    const char *fmt1 = fmt;
+    const char *err_funcname = NULL;
+    const char *err_message = NULL;
+    const char *fmt_end = NULL;
+
+    int optional = 0;
+    int keyword_only = 0;
+    HPy_ssize_t i = 0;
+    HPy_ssize_t nkw = 0;
+    HPy current_arg;
+
+    fmt_end = parse_errstring(fmt, &err_funcname, &err_message);
+
+    // first count positional only arguments
+    while (keywords[nkw] != NULL && !*keywords[nkw]) nkw++;
```

Comment:
Do we enforce braces for blocks?

Reply:
we don't have style guidelines. Maybe we should

Reply:
Braces added!

I want to discuss style guidelines / linters on the HPy call on Thursday morning (I'd like to have them, but we need to agree as a group on what to have exactly).